src/core/vfm/metastore_test.go

```diff
@@ -0,0 +1,480 @@
+
+		nsid2, err := ms.CreateNamespace("NS 2", "Brad")
+		FatalIfErr(t, err)
+
+		FatalIfErr(t, ms.AddUserToNamespace(nsid2, "Brad", "Rico"))
+		FatalIfErr(t, ms.AddObjectToNamespace(nsid2, "Rico", void2))
+
+		ExpectErr(t, ms.RemoveObjectFromNamespace(nsid2, "Rico", void3),
+			ErrNotRoot)
+		ExpectErr(t, ms.RemoveObjectFromNamespace(nsid1, "Brad", void1),
+			ErrNoAccess)
+
+		err = ms.RemoveObjectFromNamespace(nsid2, "Rico", void2)
+		FatalIfErr(t, err)
+
+		FatalIfErr(t, ms.CheckVObjectAccess(void2, "Brad", ErrAccess, nil))
+		FatalIfErr(t, ms.CheckVObjectAccess(void3, "Brad", ErrAccess, nil))
+		FatalIfErr(t, ms.CheckVObjectAccess(void2, "Rico", nil, ErrNoAccess))
+		FatalIfErr(t, ms.CheckVObjectAccess(void3, "Rico", nil, ErrNoAccess))
+	})
+
+	t.Run("Namespace Details", func(t *testing.T) {
+		FatalIfErr(t, ms.CreateUser("Joey"))
+
+		nsid, err := ms.CreateNamespace("NS 1", "Joey")
+		FatalIfErr(t, err)
+
+		void, err := ms.CreateFreeObjectInNamespace(nsid, "Joey",
+			"Folder 1", VFM_Dir_Type)
+		FatalIfErr(t, err)
+
+		namespace, err := ms.GetNamespaceDetails(nsid, "Joey")
+		ExpectTrue(t, namespace.NSID == nsid)
+		ExpectTrue(t, namespace.Name == "NS 1")
+		ExpectTrue(t, namespace.RootObjects[0] == void)
+		ExpectTrue(t, namespace.Users[0] == "Joey")
+	})
+
+	t.Run("Object Details", func(t *testing.T) {
+		FatalIfErr(t, ms.CreateUser("Houston"))
+		FatalIfErr(t, ms.CreateUser("Dallas"))
+
+		nsid1, err := ms.CreateNamespace("NS 1", "Houston")
+		FatalIfErr(t, err)
+		FatalIfErr(t, ms.AddUserToNamespace(nsid1, "Houston", "Dallas"))
+
+		nsid2, err := ms.CreateNamespace("NS 2", "Houston")
+		FatalIfErr(t, err)
+
+		nsid3, err := ms.CreateNamespace("NS 3", "Dallas")
+		FatalIfErr(t, err)
+
+		nsid4, err := ms.CreateNamespace("NS 4", "Houston")
+		FatalIfErr(t, err)
+
+		void1, err := ms.CreateFreeObjectInNamespace(nsid1, "Houston",
+			"Folder 1", VFM_Dir_Type)
+		FatalIfErr(t, err)
+
+		void2, err := ms.CreateObject(void1, "Houston",
+			"Folder 2", VFM_Dir_Type)
+		FatalIfErr(t, err)
+
+		void3, err := ms.CreateObject(void2, "Houston", "File 1",
+			VFM_File_Type)
+
+		FatalIfErr(t, ms.AddObjectToNamespace(nsid2, "Houston", void2))
+		FatalIfErr(t, ms.AddObjectToNamespace(nsid3, "Dallas", void2))
+		FatalIfErr(t, ms.AddObjectToNamespace(nsid4, "Houston", void3))
+
+		obj1_h, err := ms.GetObjectDetails(void1, "Houston")
+		FatalIfErr(t, err)
+
+		ExpectTrue(t, obj1_h.GetID() == void1)
+		sobjects1_h, _ := obj1_h.GetSubObjects()
+		ExpectTrue(t, sobjects1_h[0].Namespaces[0].NSID == nsid2)
+		ExpectTrue(t, len(sobjects1_h[0].Namespaces) == 1)
+
+		obj2_d, err := ms.GetObjectDetails(void2, "Dallas")
+		FatalIfErr(t, err)
+
+		ExpectTrue(t, len(obj2_d.GetNamespaces()) == 2)
+		sobjects2_d, _ := obj2_d.GetSubObjects()
+		ExpectTrue(t, len(sobjects2_d[0].Namespaces) == 0)
+	})
+
+	// Big tests...
+	t.Run("Big Test 1", BigTest1(ms))
+
+	FatalIfErr(t, ms.Destruct())
+}
+
+func BigTest1(ms *MetaStore) func(t *testing.T) {
```

Comment:
What is this function testing, specifically? One of the rules of testing is that a test should only ever test a single behavior.

Reply:
Well, maybe this is conventionally incorrect, but I wanted a test that ran a series of random operations on a larger file structure. All of my previous unit tests isolate specific behaviors and thus are quite small.

I wanted to make sure that no strange errors occurred when working with a larger file system.

Reply:
It's fine to keep, since more tests are certainly better than less tests. Generally tests like that last one are integration tests, but they still are generally testing a specific interaction between the two SUTs.